pkgs/development/interpreters/python/mk-python-derivation.nix

```diff
@@ -179,7 +176,25 @@ let
     # If given use the specified checkPhase, otherwise use the setup hook.
     # Longer-term we should get rid of `checkPhase` and use `installCheckPhase`.
     installCheckPhase = attrs.checkPhase;
-  }));
+  } // lib.optionalAttrs (python.stdenv.hostPlatform != python.stdenv.buildPlatform)
+    (with python.stdenv.hostPlatform; let
+    # See: https://github.com/python/cpython/blob/e488e300f5c01289c10906c2e53a8e43d6de32d8/configure.ac#L724
+    pythonArch =
+      if isAarch32 then
+        if parsed.cpu.significantByte.name == "littleEndian" then "arm" else "armeb"
+      else parsed.cpu.name;
+    pythonHostPlatform = "${pythonArch}-${parsed.kernel.name}-${parsed.abi.name}";
+  in {
+    # Python packages built through cross-compilation are always for the host platform.
+    disallowedReferences = [ python.pythonForBuild ];
+    # Make distutils compile extension modules against host Python
+    _PYTHON_HOST_PLATFORM = pythonHostPlatform;
+    _PYTHON_SYSCONFIGDATA_NAME = "_sysconfigdata__${parsed.kernel.name}_${pythonHostPlatform}";
+    # Allow host Python's sysconfigdata to be imported
+    preBuild = ''
+      export PYTHONPATH="${python}/lib/${python.libPrefix}:$PYTHONPATH"
```

Comment:
Again, maybe in a setup hook. Normally this path is implicitly added to `sys.path`, but this is needed here because we are mixing build and host libraries.

Reply:
These indeed belong in the setup hook.

But, adding the host `lib/pythonx.y` to `PYTHONPATH` seems tricky, because it also includes other native modules. Would it be wiser to copy the `_sysconfigdata__*.py` to another derivation/output and adding only that one?

Reply:
I think it is safe to add this path to `PYTHONPATH` because all the native modules have platform specific suffixes, which cpython seems to use to decide which to load. That's how `cryptography` is able to work with both host and build `cffi` on `PYTHONPATH`.

Reply:
I found a problem with this: the `subprocess` module hardcodes a path to bash, and if this comes first it tries to invoke host bash. Moving it to the end of `PYTHONPATH` should probably fix it.

Reply:
Actually, that's wrong because `PYTHONPATH` is prepended to `sys.path`. I guess we need to add sysconfigdata to its own derivation like you suggested.